packages/jobs/lib/crons/autoIdleDemo.ts

```diff
@@ -0,0 +1,128 @@
+import { schedule } from 'node-cron';
+import {
+    CommandToActivityLog,
+    ErrorSourceEnum,
+    ScheduleStatus,
+    SyncClient,
+    SyncCommand,
+    SyncConfig,
+    createActivityLog,
+    createActivityLogMessageAndEnd,
+    db,
+    errorManager,
+    updateSuccess as updateSuccessActivityLog,
+    updateScheduleStatus,
+    isErr
+} from '@nangohq/shared';
+import tracer from 'dd-trace';
+
+export async function cronAutoIdleDemo(): Promise<void> {
+    schedule('1 * * * *', () => {
+        const span = tracer.startSpan('cron.syncs.idleDemo');
+        tracer.scope().activate(span, async () => {
+            try {
+                await exec();
+            } catch (err: unknown) {
+                const e = new Error('failed_to_auto_idle_demo', { cause: err instanceof Error ? err.message : err });
+                errorManager.report(e, { source: ErrorSourceEnum.PLATFORM }, tracer);
+            }
+            span.finish();
+        });
+    });
+}
+
+interface ResQuery {
+    id: string;
+    name: string;
+    environment_id: number;
+    provider: string;
+    connection_id: number;
+    unique_key: string;
+    schedule_id: string;
+}
+async function exec() {
+    console.log('[autoidle] starting');
+
+    const q = db.knex
+        .queryBuilder()
+        .withSchema(db.schema())
+        .from<SyncConfig>('_nango_syncs')
+        .select(
+            '_nango_syncs.id',
+            '_nango_syncs.name',
+            '_nango_connections.environment_id',
+            '_nango_configs.provider',
+            '_nango_configs.unique_key',
+            '_nango_connections.connection_id',
+            '_nango_sync_schedules.schedule_id'
+        )
+        .join('_nango_connections', '_nango_connections.id', '_nango_syncs.nango_connection_id')
+        .join('_nango_environments', '_nango_environments.id', '_nango_connections.environment_id')
+        .join('_nango_configs', function () {
+            this.on('_nango_configs.environment_id', '_nango_connections.environment_id').on(
+                '_nango_configs.unique_key',
+                '_nango_connections.provider_config_key'
+            );
+        })
+        .join('_nango_sync_schedules', '_nango_sync_schedules.sync_id', '_nango_syncs.id')
+        .where({
+            '_nango_syncs.name': 'github-issues-lite',
+            '_nango_environments.name': 'dev',
+            '_nango_configs.unique_key': 'demo-github-integration',
+            '_nango_configs.provider': 'github',
+            '_nango_syncs.deleted': false,
+            '_nango_sync_schedules.status': ScheduleStatus.RUNNING
+        })
+        .where(db.knex.raw("_nango_syncs.created_at <  NOW() - INTERVAL '25h'"));
```

Comment:
should it be `.updated_at` otherwise if a demo sync is somehow unpaused. It would be paused again immediately.

Reply:
true 👌🏻

Reply:
fixed in [e6c2681](https://github.com/NangoHQ/nango/pull/1631/commits/e6c2681df513185d4e4f39587c7afb7980aa8557)